srtcore/core.cpp

```diff
@@ -8723,11 +8723,14 @@
     // srt_recvfile (which doesn't make any sense), you'll have a deadlock.
     if (m_config.bDriftTracer)
     {
+        //enterCS(m_RcvBufferLock);
```

Comment:
I am also not sure why `m_GlobControlLock` is locked here.
But `m_RcvBufferLock` protection is required.
```c++
m_pRcvBuffer->addRcvTsbPdDriftSample(ctrlpkt.getMsgTimeStamp(), tsArrival, rtt);
```

Reply:
Need to refresh myself why the `m_GlobControlLock` was added there in case of bonding only.

But the call to `addRcvTsbPdDriftSample` redirects to a call on the inside `m_tsbpd` object's `addDriftSample`, which uses its own mutex protection on the main object. No other fields in the buffer are accessed.

Reply:
Well, `m_GlobControlLock` is there likely because at the moment when one socket gets updated the drift, it must be prevented that any other member socket gets the drift updated simultaneously, and this lock must last until the drift is synchronized with all other members. This is maybe excessive and better ways of limiting the cases of locking here could be elaborated, but this whole problem will disappear anyway when the common receiver buffer for groups is implemented.

Reply:
> But the call to `addRcvTsbPdDriftSample` redirects to a call on the inside `m_tsbpd` object's `addDriftSample`, which uses its own mutex protection on the main object. No other fields in the buffer are accessed.

It is more a state protection of the RCV buffer. Adding a drift sample may change the time base, while another thread ca be in `CRcvBuffer::readMessag` where the time base is used (updated in case of a carry over) in a for loop.
Well, maybe there is no side effect of not locking the buffer here 🤔

Reply:
The commented-out code has to be resolved. Better to transform it into a comment describing that `m_RcvBufferLock` is not required because TSBPD time updates are protected internally.